src/coreclr/src/jit/lowerarmarch.cpp

```diff
@@ -535,8 +535,255 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
+        // insert the non-constant values which should still allow some gains.
+
+        return;
+    }
+    else if (argCnt == 1)
+    {
+        // We have the following (where simd is simd8 or simd16):
+        //          /--*  op1  T
+        //   node = *  HWINTRINSIC   simd   T Create
+
+        // We will be constructing the following parts:
+        //           /--*  op1  T
+        //   node  = *  HWINTRINSIC   simd   T DuplicateToVector
+
+        // This is roughly the following managed code:
+        //   return AdvSimd.Arm64.DuplicateToVector(op1);
+
+        if (varTypeIsLong(baseType) || (baseType == TYP_DOUBLE))
+        {
+            node->gtHWIntrinsicId =
+                (simdType == TYP_SIMD8) ? NI_AdvSimd_Arm64_DuplicateToVector64 : NI_AdvSimd_Arm64_DuplicateToVector128;
+        }
+        else
+        {
+            node->gtHWIntrinsicId =
+                (simdType == TYP_SIMD8) ? NI_AdvSimd_DuplicateToVector64 : NI_AdvSimd_DuplicateToVector128;
+        }
+        return;
+    }
+
+    // We have the following (where simd is simd8 or simd16):
+    //          /--*  op1 T
+    //          +--*  ... T
+    //          +--*  opN T
+    //   node = *  HWINTRINSIC   simd   T Create
+
+    if (op1->OperIsList())
+    {
+        argList = op1->AsArgList();
+        op1     = argList->Current();
+        argList = argList->Rest();
+    }
+
+    // We will be constructing the following parts:
+    //          /--*  op1  T
+    //   tmp1 = *  HWINTRINSIC   simd8  T CreateScalarUnsafe
+    //   ...
+
+    // This is roughly the following managed code:
+    //   var tmp1 = Vector64.CreateScalarUnsafe(op1);
+    //   ...
+
+    NamedIntrinsic createScalarUnsafe =
+        (simdType == TYP_SIMD8) ? NI_Vector64_CreateScalarUnsafe : NI_Vector128_CreateScalarUnsafe;
+
+    tmp1 = comp->gtNewSimdHWIntrinsicNode(simdType, op1, createScalarUnsafe, baseType, simdSize);
+    BlockRange().InsertAfter(op1, tmp1);
+    LowerNode(tmp1);
+
+    unsigned N   = 0;
+    GenTree* opN = nullptr;
+
+    for (N = 1; N < argCnt - 1; N++)
+    {
+        // We will be constructing the following parts:
+        //   ...
+        //   idx  =    CNS_INT       int    N
+        //          /--*  tmp1 simd
+        //          +--*  idx  int
+        //          +--*  opN  T
+        //   tmp1 = *  HWINTRINSIC   simd   T Insert
+        //   ...
+
+        // This is roughly the following managed code:
+        //   ...
+        //   tmp1 = AdvSimd.Insert(tmp1, N, opN);
+        //   ...
+
+        opN = argList->Current();
+
+        idx = comp->gtNewIconNode(N, TYP_INT);
+        BlockRange().InsertBefore(opN, idx);
+
+        tmp1 = comp->gtNewSimdHWIntrinsicNode(simdType, tmp1, idx, opN, NI_AdvSimd_Insert, baseType, simdSize);
+        BlockRange().InsertAfter(opN, tmp1);
+        LowerNode(tmp1);
+
+        argList = argList->Rest();
+    }
+
+    assert(N == (argCnt - 1));
```

Comment:
It seems like it would be more straightforward to include this last case in the loop above - unless I'm missing something it's identical to all the other cases, except in the case where there are 2 args, and checking for that inside the loop seems pretty low cost and quite a bit cleaner.

Reply:
Its different because it modifies the original node, rather than introducing a new node.

Reply:
Ah, makes sense.